### ISSUE TITLE
Make preview window non-focusable

### ISSUE THREAD

**Issue by 93-infinity** [QL-Win/QuickLook#644]
In Windows Explorer, the arrow keys navigate icons. This behaviour is retained using QuickLook, as expected. Pressing the arrow keys while QuickLook is open switches the preview to the newly selected icon

However if a video is previewed, and a part of the QuickLook UI is clicked (seeking, mute button, etc), focus is now transferred to the QuickLook window. Now, the arrow keys no longer work to switch items because QuickLook has focus, not windows explorer. The only solution is to either click the explorer window to give it focus (however, QuickLook will now be under the explorer window unless always on top is enabled), or to exit the preview by tapping space, then re-entering it by tapping space again, before the arrow keys work again.

Basically QuickLook steals focus when the window is clicked and the arrow keys no longer work to switch items. In macOS QuickLook does not steal focus, the uppermost window remains uppermost even if the QuickLook UI is clicked.

**Comment by xupefei** [comment]
There is another issue - sorry I can't find it now - discussing whether QL should steal focus.
There are two solutions:

1. Make QL non-focusable at all, even by clicking the preview window.
2. Pass keypresses back to the underlying window.

Solution 1 is very simple to implement and *was* exactly how QL behaviours ~2y ago. Then changed it to the current behaviour since CTRL-C in TextViewer does not work when the preview is non-focusable.

My idea is to bring the old behaviour back and add a context menu (for copying selected text") to the TextViewer.

**Comment by rabelux** [comment]
It was #338

**Comment by xupefei** [comment]
I have modified the code according to Solution 1: [[download]](https://www.icloud.com/iclouddrive/0oMRhYtBpSfnRvKBXKSpiWeKQ#Debug-no-focus)       Code: https://github.com/QL-Win/QuickLook/commit/472ccc117594b3170ed80acb362d20ee2e1eee5e and [WindowHelper](https://github.com/QL-Win/QuickLook.Common/blob/master/Helpers/WindowHelper.cs)

The only bug I found is about the window z-order: if you switch between multiple Explorer windows and manage to make QL window not top-most, you are unable to bring it back to top-most by clicking in the title bar area. This is because that, clicking on the title bar does not generate a MouseDown event hence our `PreviewMouseDown` listener didn't get fired.

I could think of two solutions for this bug:

1. Make the preview topmost, i.e., force <img width="43" alt="image" src="https://user-images.githubusercontent.com/1687847/81343292-68d17e80-90bd-11ea-9ab8-5a53281df087.png">
2. Switch to Solution 2

Any comments?

**Comment by rabelux** [comment]
Can this be tested by downloading the latest nightly? Or does the nightly only reflect changes made in the master-branch

**Comment by xupefei** [comment]
@rabelux Here it is: https://ci.appveyor.com/project/xupefei/quicklook/builds/32856390/artifacts

**Comment by rabelux** [comment]
Thanks!
It's all looking good so far, just came across 2 issues:
1. When using the trackpad of my laptop scrolling/zooming is possible, but not with a (USB-)connected mouse. Any idea what could cause this?
1. Not really a bug, just a minor inconvenience:
The dragging-behaviour feels broken, as it doesn't show the frame of the original window

**Comment by simioni** [comment]
![version_info](https://i.imgur.com/TgLiYwd.jpg)
@rabelux Can you confirm you're running **v3.6.7-16-g1cd5019**? Did you install from the MSI or are running directly from the .exe in the zip archive? What type of file are you previewing when you perceive the bug?

I could not reproduce either of these issues. Zooming / scrolling is working fine with trackpad and mouse (and touchscreen) for me. Dragging the QuickLook window also appears to be working correctly.

**Comment by xupefei** [comment]
@rabelux for the second issue: it might be related to your GPU driver. Could you upload a screen-recording?

**Comment by rabelux** [comment]
@simioni Confirming the version and I used the MSI-installer.
![image](https://user-images.githubusercontent.com/16120882/81968905-5dcaa100-961d-11ea-998a-446901254e75.png)
The bug appears on actually any format. I have tried with txt, pdf, xlsx, jpg, rtf, bak and png so far.

**Comment by xupefei** [comment]
Are you using Windows 8? Then you might have this scrolling issue because 
1) Prior to Windows 10, there is no "scroll window under cursor" feature: https://www.nextofwindows.com/turn-onoff-scroll-inactive-window-under-mouse-cursor-in-windows-10
2) When "scroll window under cursor" is off, a window must be focused to receive mouse wheel events
3) Now QL window is non-focusable

I think I should disable this "non-focusable" behaviour for Windows 7 and 8/8.1.

**Comment by rabelux** [comment]
I'm still on Win 7. That explains why scrolling with the mouse is not working. But why does the trackpad work? Is that some magic the synaptics driver does?
And is the dragging-issue connected to Win 7 not being able to handle non-focussed windows?
I just installed the last official release (v3.6.7) and dragging works fine.
Dragging with old version:
![oldver](https://user-images.githubusercontent.com/16120882/81992550-02f87000-9644-11ea-9849-861bad8e6ffd.gif)
Dragging with new version:
![dragissue](https://user-images.githubusercontent.com/16120882/81992482-d3e1fe80-9643-11ea-98c8-29490a31d02d.gif)

**Comment by xupefei** [comment]
I see. Let me grab a Windows 8 VM and try again.

**Comment by xupefei** [comment]
Fixed!

**Comment by xupefei** [comment]
I have to revert this change since, as you may have discovered, Windows has some bugs regarding the non-focusable window. The Z-order becomes messy when you do the follows:

1. In an explorer window, select a file and preview it.
2. Click on the title bar of the preview window.
3. Go back to the explorer window and press up/down arrow keys.

After step 3, you will find the up/down arrow keys no longer change the selected file. Meanwhile, the focus switched to nowhere unless you close the preview window.

I will investigate this issue later, but now I have to rollback the change.

**Comment by hansbizarre** [comment]
If anyone is still looking for a workaround to this, you can install AutoHotkey (free) and use the script in this thread:
https://www.autohotkey.com/boards/viewtopic.php?f=82&t=121811

**Comment by emako** [comment]
Currently nightly build version has canceled focusable window
